### ISSUE TITLE
ebay-notice: not recognized as landmark

### ISSUE THREAD

**Issue by mkchang** [eBay/ebayui-core#1198]
# Bug Report

## Description
ebay-notice is not recognized as a landmark, at least with VO + Safari on Mac. To test, use Ctrl-Opt-U to turn on macOS VoiceOver Rotor and use arrow keys to select 'Landmarks' mode, and see if the notices show up in the list of landmarks. This is a follow up to the issue that originally caused #1185 to be opened.

Unlike in the [Skin notice docs](https://ebay.github.io/skin/#notice), in my app, `aria-hidden` is applied to the icon, which seems to hide the `aria-label` that `ebay-notice` passes to `ebay-icon`, and prevents it from being detected as a landmark. It works as long as I delete that `aria-hidden`. Not sure how to reconcile that, unless `ebay-notice` should treat the icon as a content image, assuming you are providing an aria-label. 
```html
<section role="region" class="page-notice page-notice--attention" aria-labelledby="s0-0-15-2-2-1[0]-1-status">
    <h2 class="page-notice__status" id="s0-0-15-2-2-1[0]-1-status">
        <svg class="icon icon--attention-filled" focusable="false" aria-label="attention" aria-hidden="true">
            <use xlink:href="#icon-attention-filled"></use>
        </svg>
    </h2>
    <div class="page-notice__content">
        This notice is not recognized as a landmark
    </div>
</section>

<section class="page-notice page-notice--attention" role="region" aria-labelledby="attention-status">
    <h4 class="page-notice__status" id="attention-status">
        <svg class="icon icon--attention-filled" focusable="false" height="24" width="24" aria-label="Attention">
            <use xlink:href="#icon-attention-filled"></use>
        </svg>
    </h4>
    <div class="page-notice__content">
        This notice is recognized as a landmark
    </div>
</section>
```

Also explored from our discussion about `ebay-notice` passing `ebay-icon` `a11Text` instead of setting an `aria-label`. This resulted in this HTML for me, but it wasn't recognized as a landmark either, thinking `title` + `aria-labelledby` does not work for landmark detection.
```html
<section role="region" class="page-notice page-notice--confirmation" aria-labelledby="s0-0-15-2-2-1[1]-1-status">
    <h2 class="page-notice__status" id="s0-0-15-2-2-1[1]-1-status">
        <svg class="icon icon--confirmation-filled"
            focusable="false" aria-labelledby="s0-0-15-2-2-1[1]-1-5-text" role="img">
            <title id="s0-0-15-2-2-1[1]-1-5-text">Test title</title>
            <use xlink:href="#icon-confirmation-filled"></use>
        </svg>
    </h2>
    <div class="page-notice__content">
        This notice is recognized as a landmark
    </div>
</section>
```
## Workaround
Deleting just the `aria-hidden` does seem to work for me, whereas passing `a11yText` to `ebay-icon` instead of setting `aria-label` does not.

**Comment by ianmcburnie** [comment]
Thanks @mkchang for the detailed research into this issue. The notice component is undergoing some big changes in v6 but we will see if we can get this fixed in v5 also.

**Comment by mkchang** [comment]
Thanks guys! Tried out 5.6.4, this does seem to fix VO announcing the content when focused on the icon (and even allowing focus on the icon instead of just treating it as an empty h2), but it still leaves the notice as a whole not recognized as a landmark. I think it's because in #1206, by using `a11yText`, it results in the `title` + `aria-labelledby` combination, which for me does not work for landmark detection in Safari+VO.

**Comment by ianmcburnie** [comment]
@agliga I'm not sure why this isn't just following Skin markup...(from v10 branch)?

**Comment by agliga** [comment]
The issue is because of the way ebay icon is implemented. Notice calls `ebay-icon` to set the icon correctly.

`ebay-icon` requires an `a11y-text`. If it does not have it, it sets `aria-hidden=true`. See: https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-icon/index.marko#L55-L57

`a11y-text` sets `<title>` which is linked to the icon by `aria-labelledby`: https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-icon/index.marko#L55-L57  

As to why it is implemented like this, I am not sure, this was before my time. But it might be a bug in icon if we should support `aria-label` as well.

**Comment by ianmcburnie** [comment]
@agliga Please take a look and see what it would take to also support aria-label in icon, and leverage that instead of title. It's fine if an icon has both an aria-label and a title (so long as they are the same value). If it makes it even more convoluted, let's just fix this in v6 (given that there is a workaround in the app).

**Comment by ianmcburnie** [comment]
Fixed in 5.7.0. Closing.
